src/main/java/rapgod/utils/Parser.java

```diff
@@ -0,0 +1,92 @@
+package rapgod.utils;
+
+import rapgod.tasks.Task;
+import rapgod.tasks.ToDo;
+import rapgod.tasks.Deadline;
+import rapgod.tasks.Event;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Parser {
+
+    /**
+     * Parses the input string to a LocalDateTime if the input contains both date and time.
+     * If the input contains only a date, a default time of 0000 is appended and parsed.
+     * Supports multiple date and time formats.
+     *
+     * @param input The input string to be parsed.
+     * @return A LocalDateTime object representing the parsed date and time.
+     * @throws IllegalArgumentException If the input format is invalid.
+     */
+    public static LocalDateTime parseToDateTime(String input) {
+        DateTimeFormatter dateTimeFormatter1 = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
+        DateTimeFormatter dateTimeFormatter2 = DateTimeFormatter.ofPattern("MMM dd yyyy");
+        DateTimeFormatter dateTimeFormatter3 = DateTimeFormatter.ofPattern("MMM dd yyyy h:mma");
+        DateTimeFormatter dateTimeFormatter4 = DateTimeFormatter.ofPattern("MMM dd yyyy hh:mma");
```

Comment:
Maybe don't use numbers to distinguish names, I suggest making the names more descriptive like dateTimeFormatterSlashed

Reply:
Thanks for all the comments Isaac! Will rectify the issues